.github/workflows/style.yaml

```
@@ -2,65 +2,84 @@ name: style
+      - name: Set up python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: "3.12"
+      - name: Cache dependencies
+        uses: actions/cache@v4.0.2
+        id: cache
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('requirements/required.txt') }}-${{ hashFiles('requirements/datasets.txt') }}-${{ hashFiles('requirements/style.txt') }}-${{ hashFiles('requirements/tests.txt') }}
+      - name: Install pip dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          pip install -r requirements/required.txt -r requirements/datasets.txt -r requirements/style.txt -r requirements/tests.txt
+          pip cache purge
+      - name: List pip dependencies
+        run: pip list
+      - name: Run mypy checks
+        run: mypy .
   ruff:
     name: ruff
     runs-on: ubuntu-latest
     steps:
-    - name: Clone repo
-      uses: actions/checkout@v4.1.4
-    - name: Set up python
-      uses: actions/setup-python@v5.1.0
-      with:
-        python-version: '3.12'
-    - name: Cache dependencies
-      uses: actions/cache@v4.0.2
-      id: cache
-      with:
-        path: ${{ env.pythonLocation }}
-        key: ${{ env.pythonLocation }}-${{ hashFiles('requirements/style.txt') }}
-    - name: Install pip dependencies
-      if: steps.cache.outputs.cache-hit != 'true'
-      run: |
-        pip install -r requirements/style.txt
-        pip cache purge
-    - name: List pip dependencies
-      run: pip list
-    - name: Run ruff checks
-      run: |
-        ruff check --output-format=github --no-fix .
-        ruff format --diff
+      - name: Clone repo
+        uses: actions/checkout@v4.1.2
+      - name: Set up python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: "3.12"
+      - name: Cache dependencies
+        uses: actions/cache@v4.0.2
+        id: cache
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('requirements/style.txt') }}
+      - name: Install pip dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          pip install -r requirements/style.txt
+          pip cache purge
+      - name: List pip dependencies
+        run: pip list
+      - name: Run ruff checks
+        run: |
+          ruff check --output-format=github --no-fix .
+          ruff format --diff
+  prettier:
+    name: prettier
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@v4.1.2
+      - name: Set up nodejs
+        uses: actions/setup-node@v4.0.2
+        with:
+          node-version: "20"
+          cache: "npm"
+      - name: Installing prettier
+        run: |
+          npm ci
+          npm cache clean --force
+      - name: List npm dependencies
+        run: npm ls --all
+      - name: Run prettier formatting
+        run: npx prettier . --write
```

```suggestion
        run: npx prettier . --check
```
Write is used to format, check is used to confirm that everything looks good. The former is used by users, but the latter is used in CI.

---

True, my mistake.